code/iaas/lb-instance/src/main/java/io/cattle/platform/lb/instance/dao/impl/LoadBalancerInstanceDaoImpl.java

```diff
@@ -0,0 +1,55 @@
+package io.cattle.platform.lb.instance.dao.impl;
+
+import static io.cattle.platform.core.model.tables.NetworkTable.NETWORK;
+import io.cattle.platform.core.constants.NetworkConstants;
+import io.cattle.platform.core.dao.AccountDao;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.dao.NetworkDao;
+import io.cattle.platform.core.model.LoadBalancer;
+import io.cattle.platform.core.model.LoadBalancerHostMap;
+import io.cattle.platform.core.model.Network;
+import io.cattle.platform.db.jooq.dao.impl.AbstractJooqDao;
+import io.cattle.platform.lb.instance.dao.LoadBalancerInstanceDao;
+import io.cattle.platform.object.ObjectManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class LoadBalancerInstanceDaoImpl extends AbstractJooqDao implements LoadBalancerInstanceDao {
+
+    @Inject
+    GenericMapDao mapDao;
+
+    @Inject
+    ObjectManager objectManager;
+
+    @Inject
+    NetworkDao ntwkDao;
+
+    @Inject
+    AccountDao accountDao;
+
+    @Override
+    public List<Long> getLoadBalancerHosts(long lbId) {
+        List<Long> hostIds = new ArrayList<Long>();
+        for (LoadBalancerHostMap map : mapDao.findToRemove(LoadBalancerHostMap.class, LoadBalancer.class, lbId)) {
+            hostIds.add(map.getHostId());
+        }
+        return hostIds;
+    }
+
+    @Override
+    public Network getLoadBalancerInstanceNetwork(LoadBalancer loadBalancer) {
+        // today the only network in cattle is owned by the system account, so passing its id
```

Comment:
Can you just check for the account owned one and if none is found go to the system account?